### PR TITLE
remove experimental implicit deref

### PR DESCRIPTION
**Pull request by haxscramper**
Remove implementation and documentation for the experimental implicit deref. The feature is removed largely because it is not used anywhere, not tested (the only two existing tests have been disabled for the last two years in `when false`) block and most importantly have subpar implementation.

### CODE REVIEW

**Comment by haxscramper** [comment]
500th PR 🚀🎉

File: tests/implicit/timplicit.nim [hunk 1]
```diff
@@ -29,21 +27,3 @@ var sp: PPValue = addr x
 sp.a = 2
 if sp.a == 2: echo 2  # with sp[].a the error is gone
```
Comment (saem) [inline tests/implicit/timplicit.nim]: Is this test really testing anything anymore? Maybe we can kill the whole file.

**Review by saem — COMMENTED** [review]
Noticed one thing, not a blocker. Thanks.

**Pull request merged**